plugins/backend-ekg/src/Cardano/BM/Backend/EKGView.lhs

```diff
@@ -216,7 +233,10 @@ instance (ToJSON a, FromJSON a) => IsBackend EKGView a where
         evref <- newEmptyMVar
         let ekgview = EKGView evref
         evport <- getEKGport config
-        ehdl <- forkServer "127.0.0.1" evport
+        ehdl <- (forkServer "127.0.0.1" evport
+                 -- This unfortunate delay is to catch the async exception.
+                 <* threadDelay 300000)
```

Comment:
@dcoutts, asking for advice -- what would be the proper way to deal with the async exception unfortunately thrown by `forkServer` (which we don't control)?

Reply:
The main problem is that it's a long-running thread, with no communication API -- so we can't ask it if it has succeeded or not, and especially wait for such..

Reply:
Yes, unfortunately `forkServer` does not make it easy for us. We can run it in its own thread so that the async exception that `forkServer` throws will go to a thread we control, but this is also not easy because there's nothing to wait on.

So a temporary hack would be either what you have there, or run it in its own thread and threadDelay there, catching any async exceptions, re-wrapping them and re-throwing.

But better longer term is file a PR upstream to expose the synchronous version `startServer` that just runs the server in the current thread. Then we can fork and run it, but we then get to write the exception handler.

Reply:
So I think the most efficient combination of actions would be:

1. leave the `threadDelay` as is, _and_
2. ~file a bug upstream~ there's already https://github.com/tibbe/ekg/issues/50

Reply:
Created https://github.com/tibbe/ekg/pull/78